build-system/tasks/extension-helpers.js

```diff
@@ -561,18 +561,112 @@ async function buildExtensionJs(path, name, version, latestVersion, options) {
-    fs.copyFileSync(dir + 'worker.js', `${file}.js`);
-    // The "mjs" output is unminified ES6 and has debugging flags enabled.
-    fs.copyFileSync(dir + 'worker.mjs', `${file}.max.js`);
+    copyWorkerDomResources(version);
+  }
+}
 
+async function copyWorkerDomResources(version) {
+  const workerDomDir = 'node_modules/@ampproject/worker-dom';
+  const newDirExists = fs.existsSync(`${workerDomDir}/dist/amp-production`);
+  const targetDir = 'dist/v0';
+
+  log(
+    green('Copying @ampproject/worker-dom resources:'),
+    cyan('worker.js, worker.nodom.js')
+  );
+
+  if (newDirExists) {
+    const dir = `${workerDomDir}/dist`;
+    const workerFilesToDeploy = new Map([
+      [
+        `${dir}/amp-production/worker/worker.js`,
+        `${targetDir}/amp-script-worker-${version}.js`,
+      ],
+      [
+        `${dir}/amp-production/worker/worker.nodom.js`,
+        `${targetDir}/amp-script-worker-nodom-${version}.js`,
+      ],
+      [
+        `${dir}/amp-production/worker/worker.mjs`,
+        `${targetDir}/amp-script-worker-${version}.mjs`,
+      ],
+      [
+        `${dir}/amp-production/worker/worker.nodom.mjs`,
+        `${targetDir}/amp-script-worker-nodom-${version}.mjs`,
+      ],
+      [
+        `${dir}/amp-production/worker/worker.js.map`,
+        `${targetDir}/amp-script-worker-${version}.js.map`,
+      ],
+      [
+        `${dir}/amp-production/worker/worker.nodom.js.map`,
+        `${targetDir}/amp-script-worker-nodom-${version}.js.map`,
+      ],
+      [
+        `${dir}/amp-production/worker/worker.mjs.map`,
+        `${targetDir}/amp-script-worker-${version}.mjs.map`,
+      ],
+      [
+        `${dir}/amp-production/worker/worker.nodom.mjs.map`,
+        `${targetDir}/amp-script-worker-nodom-${version}.mjs.map`,
+      ],
+      [
+        `${dir}/amp-debug/worker/worker.js`,
+        `${targetDir}/amp-script-worker-${version}.max.js`,
+      ],
+      [
+        `${dir}/amp-debug/worker/worker.nodom.js`,
+        `${targetDir}/amp-script-worker-nodom-${version}.max.js`,
+      ],
+      [
+        `${dir}/amp-debug/worker/worker.mjs`,
+        `${targetDir}/amp-script-worker-${version}.max.mjs`,
+      ],
+      [
+        `${dir}/amp-debug/worker/worker.nodom.mjs`,
+        `${targetDir}/amp-script-worker-nodom-${version}.max.mjs`,
+      ],
+      [
+        `${dir}/amp-debug/worker/worker.js.map`,
+        `${targetDir}/amp-script-worker-${version}.max.js.map`,
+      ],
+      [
+        `${dir}/amp-debug/worker/worker.nodom.js.map`,
+        `${targetDir}/amp-script-worker-nodom-${version}.max.js.map`,
+      ],
+      [
+        `${dir}/amp-debug/worker/worker.mjs.map`,
+        `${targetDir}/amp-script-worker-${version}.max.mjs.map`,
+      ],
+      [
+        `${dir}/amp-debug/worker/worker.nodom.mjs.map`,
+        `${targetDir}/amp-script-worker-nodom-${version}.max.mjs.map`,
+      ],
+    ]);
+    for (const [src, dest] of workerFilesToDeploy) {
+      await fs.copy(src, dest);
```

Comment:
Since `dir` and `targetDir` are common across all entries, how about removing them from the map and changing this line as follows?

```js
await fs.copy(`${dir}/${src}`), `${targetDir}/${dest}`);
```

Reply:
done.